File: javascript/QuickSort.js
quickSort = (array) => {
    if (array.length < 2) {
        return array;
    }
    const pivot = array[0];
    const low = array.filter(value => {
        return value < pivot;
    })
    const high = array.filter(value => {
        return value > pivot;
    })
    console.log(low);
    console.log(high)

    return [...quickSort(low), pivot, ...quickSort(high)];
}
console.log(quickSort([4, 2, 7, 1, 44, 222, 99]));

File: Python/quickSort.py
def quickSort(array):
    if len(array) < 2: return array
    else:
        # choose a pivot
        pivot = array[0];
        less = [i for i in array[1:] if i <=pivot]
        greater = [i for i in array[1:] if i > pivot]

        return quickSort(less) + [pivot] + quickSort(greater)

print(quickSort([6,4,99,12,1]))
